server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -59,16 +65,22 @@ public static Ad convertRowToAd(String[] row) {
     return ad; 
   }
 
-  /* 
-   * Min and max impression fields contain letters k and M as well as ≤. The 
-   * formatImpressionField helper method makes the fields easier to parse later.
-   */ 
-  public static String[] formatImpressionField(String str) {
-    String impressionsString = str.replace("k", "000").replace("M", "000000").replace(" ","").replace("≤", "");
-    String[] impressionsArray = impressionsString.split("-");
-    return impressionsArray; 
+  /*
+   * Helper function tries to convert date strings to LocalDates to ensure they are 
+   * in isostring format.
+   * checkDateFormat does *not* return the parsed LocalDate because Firebase does not
+   * support that datatype. Dates are converted after retrieval from Firebase.
+   */
+  public static String checkDateFormat(String date) {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    LocalDate localDate = LocalDate.parse(date, formatter);
+    return date;
   }
 
+  /*
+   * getImpressionsMin tries to parse the minimum number of impressions and throws an error
+   * if the field is not parsable.
+   */ 
   public static long getImpressionsMin(String impressionsField) throws IllegalArgumentException, NumberFormatException {
```

Comment:
It could be helpful to catch these errors and print out what is being passed in to cause the error instead of just throwing one. maybe some `try` and `catch` statements. Here and elsewhere in this file.